### DISCUSSION TITLE
Support standard HTML encodings

### DISCUSSION THREAD

**Discussion by stasoid** [litehtml/litehtml#304] [Ideas]
I find it really weird that gumbo (and, by extension, litehtml) doesn't support standard HTML encodings. These encodings are an integral part of HTML parser. It is strange that litehtml is fully capable of parsing any HTML, *except* for this part. I would like to add support for them to litehtml.

The only possible problem I see is that conversion tables for east-asian encodings may take a lot of space compared to the rest of litehtml. But before investigating this further, I would like to know whether you're opposed to this idea in principle or not?

Note that the list of standard encodings is final and will not be expanded in the future.

**Update**: east-asian tables take 216KB, litehtml is currently a little under 1MB. If this is not acceptable I can try to compress them. Should I?

**Comment by stasoid** [comment]
I'll ask another way: how small this functionality should be to be acceptable for litehtml?

litehtml can be as small as 725KB when compiled with `-Os -flto`, probably even smaller with custom STL (measured with <code><a href=https://github.com/google/bloaty>bloaty</a> litebrowser-linux -d compileunits -n 0 -s vm</code>).

**Reply by tordex** [reply]
I think the support for HTML encodings is important. It is great if you can implement this. About size. I think 256KB is not very big addition. Also it is possible (I guess possible) to exclude from compilation big tables using defines. So if somebody need smaller size he build litehtml without some/all encoding support.

**Reply by stasoid** [reply]
Thank you for the answer, you've dispelled my worries. How did I not think about conditional compilation, it seems so obvious now.
